src/walletSdk/Customer/index.ts

```diff
@@ -0,0 +1,96 @@
+import { AxiosInstance } from "axios";
+import { Sep9InfoRequiredError, CustomerNotFoundError } from "../Exceptions";
+import {
+  CustomerInfoMap,
+  Sep12Status,
+  Sep12Type,
+  Field,
+  ProvidedField,
+  GetCustomerResponse,
+  AddCustomerResponse,
+} from "../Types";
+
+export class Sep12 {
+  private token;
+  private baseUrl;
+  private httpClient;
+  private headers;
+
+  constructor(token: string, baseUrl: string, httpClient: AxiosInstance) {
+    this.token = token;
+    this.baseUrl = baseUrl;
+    this.httpClient = httpClient;
+    this.headers = {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${this.token}`,
+    };
+  }
+
+  async getByIdAndType(id: string, type: string): Promise<GetCustomerResponse> {
```

Comment:
Maybe we can split this to `getById` and `getByType`? The get method also accept optional parameters, which were not supported by Kotlin for some reason.

Reply:
I don't think there's a case where you can GET with just id and no type right? According to sep:
```
GET [KYC_SERVER]/customer&type=<customer-type>
GET [KYC_SERVER]/customer?account=<stellar-account>&type=<customer-type>
GET [KYC_SERVER]/customer?account=<stellar-account>&memo=<memo>&type=<customer-type>
GET [KYC_SERVER]/customer?account=<muxed-account>&type=<customer-type>
GET [KYC_SERVER]/customer?id=<customer-id>&type=<customer-type>
```

what about just `getCustomer` with optional id, type, and account?

and in a jsdoc we explain what combos are needed and link to the [SEP](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0012.md#customer-get)

Reply:
```
GET [KYC_SERVER]/customer&type=<customer-type>
GET [KYC_SERVER]/customer?account=<stellar-account>&type=<customer-type>
GET [KYC_SERVER]/customer?account=<stellar-account>&memo=<memo>&type=<customer-type>
GET [KYC_SERVER]/customer?account=<muxed-account>&type=<customer-type>
GET [KYC_SERVER]/customer?id=<customer-id>&type=<customer-type>
```
I thought those were a few samples to illustrate some possible variations but not all of them, but I'm also not sure about it

I like the idea of a single `getCustomer` function with optional fields

Reply:
> what about just getCustomer with optional id, optional type, and optional account?

what if we leave only `id`, `type` and `memo` as optional params? since the other optional fields (like `account`) are deprecated

Reply:
how does this look? https://github.com/stellar/typescript-wallet-sdk/pull/71/commits/af6813b1a96eef68ecf70d638d06df74915b58d6

Reply:
looks great! Left a couple comments/suggestions